vignettes/Spectra.Rmd

```diff
@@ -250,15 +250,40 @@ pks <- peaksData(sps)
 pks[[1]]
 ```
 
-
-
 Note that we would get the same result by using the `as` method to coerce a
 `Spectra` object to a `list` or `SimpleList`:
 
 ```{r as}
 as(sps, "SimpleList")
 ```
 
+Certain backends (such as the `MsBackendMemory`) support also additional peaks
+variables. Below we define *peak annotations* for each individual peak in the
+peak matrices and assign that to a variable called `peak_anno` to the `Spectra`
+object. Importantly, for peak variables, a value needs to be assigned to each
+peak, even if it is `NA` or `empty`. In the example below we define a `list` of
+`character` vectors with arbitrary IDs/names for each peak.
+
+```{r}
+sps2 <- sps
+sps2$peak_anno <- list(c("a", NA_character_, "b", "c", "d"),
+                       c("a", "b", "c", "d", "e", "f", "g"),
+                       c("a", "b", "c", "d", "e", "f", "g", "h", "i"))
+peaksVariables(sps2)
+```
+
+We can use the `peaksData` to extract the values for these additional peak
+variable.
+
+```{r}
+## Peak annotations for the first spectrum
+peaksData(sps2, "peak_anno")[[1L]]
+## Peak annotations for the second spectrum
+peaksData(sps2, "peak_anno")[[2L]]
+```
+
+This works however **only** for `Spectra` that use a `MsBackendMemory` backend.
```

Comment:
Couldn't this be implemented by all writeable backends? (but is currently just supported by `MsBackendMemory`?